vendor/github.com/hpcloud/tail/tail.go

```diff
@@ -191,7 +201,9 @@ func (tail *Tail) close() {
 func (tail *Tail) closeFile() {
 	if tail.file != nil {
 		tail.file.Close()
+		tail.fileMtx.Lock()
 		tail.file = nil
+		tail.fileMtx.Unlock()
```

Comment:
Unfortunately I think this whole function needs to hold the lock, not just when you set `tail.file` to nil

Reply:
good catch, fixed